firewall/firewall.go

```diff
@@ -135,7 +137,10 @@ func handlePacket(pkt packet.Packet) {
 	// }
 
 	// allow local dns
-	if (pkt.Info().DstPort == 53 || pkt.Info().SrcPort == 53) && pkt.Info().Src.Equal(pkt.Info().Dst) {
+	if (pkt.Info().DstPort == 53 || pkt.Info().SrcPort == 53) &&
+		(pkt.Info().Src.Equal(pkt.Info().Dst) || // Windows redirects back to same interface
```

Comment:
Good one.
Although, shouldn't the compiler optimize accessing `pkg.Info()` often?

Reply:
Not sure it can because you're returning a pointer in `pkg.Info()` so he must always de-reference the pointer and cannot use a local copy on the stack/cache because the actual struct may be modified concurrently.